### PR TITLE
Use azcopy to download test data

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Use azcopy from download_e2e_test_data.py, add helper function for downloading azcopy.
Update download_test_data.py to use helper function.

**Motivation and Context**
Test data download is slow for the training e2e test builds which sometimes causes the build to time out. Trying to speed it up by using azcopy instead.

### CODE REVIEW

**Comment by ytaous** [comment]
Do you what's improvement with azcopy?

**Comment by askhade** [comment]
The changes in this PR also affect release pipelines (python packaging, nuget ) please run release pipelines as well to make sure this change doesn't break them.

File: tools/python/get_azcopy.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+            print("Downloading azcopy from '{}'...".format(download_url))
+            urllib.request.urlretrieve(download_url, downloaded_path)
+
+            extracted_path = os.path.join(temp_dir, "azcopy")
+            shutil.unpack_archive(downloaded_path, extracted_path)
+
+            azcopy_path = _find_azcopy(extracted_path)
+
+            os.chmod(azcopy_path, stat.S_IXUSR)
```
Comment (thiagocrepaldi) [inline tools/python/get_azcopy.py:75]: Does that work for Windows?

Comment (edgchen1) [reply tools/python/get_azcopy.py:75]: it should not have any effect on Windows
https://docs.python.org/3/library/os.html#os.chmod

Comment (snnn) [reply tools/python/get_azcopy.py:76]: BTW, we have azcopy v10 preinstalled on most of our CI build machines, except the ones that created by the training team and private to training.  So I expect this change wouldn't impact the non-training pipelines. And all the training CI build machines are Linux I guess?

Comment (edgchen1) [reply tools/python/get_azcopy.py:76]: the agents that run the training E2E tests are Linux machines. get_azcopy should allow use of an existing azcopy that is the correct version.

Comment (edgchen1) [reply tools/python/get_azcopy.py:76]: I know that the Nuget pipeline uses azcopy for downloading test data in the Mac test job, that hits this new code path. I ran that and it worked though.

**Pull request merged**